File: sdk/java/src/main/java/io/pipeops/rexec/Container.java
package io.pipeops.rexec;

import java.util.HashMap;
import java.util.Map;

/**
 * Represents a Rexec container/sandbox.
 */
public class Container {
    private String id;
    private String name;
    private String image;
    private String status;
    private String createdAt;
    private String startedAt;
    private Map<String, String> labels;
    private Map<String, String> environment;

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getImage() {
        return image;
    }

    public String getStatus() {
        return status;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    public String getStartedAt() {
        return startedAt;
    }

    public Map<String, String> getLabels() {
        return labels != null ? labels : new HashMap<>();
    }

    public Map<String, String> getEnvironment() {
        return environment != null ? environment : new HashMap<>();
    }

    public boolean isRunning() {
        return "running".equals(status);
    }

    public boolean isStopped() {
        return "stopped".equals(status);
    }
}

/**
 * Request to create a new container.
 */
class CreateContainerRequest {
    private final String image;
    private String name;
    private Map<String, String> environment;
    private Map<String, String> labels;

    /**
     * Create a request with the specified image.
     */
    public CreateContainerRequest(String image) {
        this.image = image;
    }

    public CreateContainerRequest setName(String name) {
        this.name = name;
        return this;
    }

    public CreateContainerRequest setEnvironment(Map<String, String> environment) {
        this.environment = environment;
        return this;
    }

    public CreateContainerRequest setLabels(Map<String, String> labels) {
        this.labels = labels;
        return this;
    }

    public CreateContainerRequest addEnv(String key, String value) {
        if (this.environment == null) {
            this.environment = new HashMap<>();
        }
        this.environment.put(key, value);
        return this;
    }

    public CreateContainerRequest addLabel(String key, String value) {
        if (this.labels == null) {
            this.labels = new HashMap<>();
        }
        this.labels.put(key, value);
        return this;
    }
}

File: sdk/php/src/Container.php
<?php

declare(strict_types=1);

namespace Rexec;

/**
 * Represents a Rexec container/sandbox.
 *
 * Accepts both snake_case (production API) and camelCase keys.
 */
class Container
{
    public string $id;
    public ?string $name;
    public string $image;
    public string $status;
    public ?string $createdAt;
    public ?string $startedAt;
    /** @var array<string, string> */
    public array $labels;
    /** @var array<string, string> */
    public array $environment;

    public function __construct(array $data)
    {
        $this->id = (string) self::pick($data, 'id', '');
        $this->name = self::pick($data, 'name', null);
        if ($this->name !== null) {
            $this->name = (string) $this->name;
        }
        $this->image = (string) self::pick($data, 'image', '');
        $this->status = (string) self::pick($data, 'status', '');
        $this->createdAt = self::stringOrNull(self::pick($data, 'created_at', self::pick($data, 'createdAt', null)));
        $this->startedAt = self::stringOrNull(self::pick($data, 'started_at', self::pick($data, 'startedAt', null)));
        $this->labels = is_array($data['labels'] ?? null) ? $data['labels'] : [];
        $this->environment = is_array($data['environment'] ?? null) ? $data['environment'] : [];
    }

    public function isRunning(): bool
    {
        return $this->status === 'running';
    }

    public function isStopped(): bool
    {
        return $this->status === 'stopped';
    }

    private static function pick(array $data, string $key, mixed $default): mixed
    {
        return array_key_exists($key, $data) ? $data[$key] : $default;
    }

    private static function stringOrNull(mixed $value): ?string
    {
        if ($value === null || $value === '') {
            return null;
        }
        return (string) $value;
    }
}
